src/shared/Core/Authentication/AuthenticationBase.cs

```diff
@@ -110,28 +110,44 @@ protected void ThrowIfTerminalPromptsDisabled()
             }
         }
 
-        protected bool TryFindHelperExecutablePath(string envar, string configName, string defaultValue, out string path)
+        protected bool TryFindHelperCommand(string envar, string configName, string defaultValue, out string command, out string args)
+        {
+            if (TryFindHelperPath(envar, configName, defaultValue, out string path))
+            {
+                if (path.EndsWith("dll"))
```

Comment:
```suggestion
                if (path.EndsWith(".dll"))
```

..else we'd match `"my-native-programdll"` on UNIX platforms as a `dotnet` program.